sql/core/src/test/scala/org/apache/spark/sql/execution/RowFormatConvertersSuite.scala

```diff
@@ -58,6 +58,48 @@ class RowFormatConvertersSuite extends SparkPlanTest with SharedSQLContext {
     assert(!preparedPlan.outputsUnsafeRows)
   }
 
+  test("coalesce can process unsafe rows") {
+    val plan = Coalesce(1, outputsUnsafe)
+    val preparedPlan = sqlContext.prepareForExecution.execute(plan)
+    assert(getConverters(preparedPlan).size === 1)
+    assert(preparedPlan.outputsUnsafeRows)
+  }
+
+  test("coalesce can process safe rows") {
```

Comment:
We may remove the safe rows in future, it's fine to not have these tests (for safe rows).